Index: makeinfo.c
*** orig/makeinfo.c	Sun May 23 01:29:46 1993
--- makeinfo.c	Sat Jan 29 16:00:26 1994
***************
*** 472,477 ****
--- 472,482 ----
    cm_copyright (), cm_code (), cm_samp (), cm_file (), cm_kbd (),
    cm_key (), cm_ctrl (), cm_var (), cm_dfn (), cm_emph (), cm_strong (),
    cm_cite (), cm_italic (), cm_bold (), cm_roman (), cm_title (), cm_w (),
+ 
+ #ifdef JOKES
+   cm_email (), cm_explain (), cm_fyi (), cm_host (), cm_news (),
+ #endif
+ 
    cm_refill (), cm_titlefont ();
  
  int
***************
*** 606,611 ****
--- 611,625 ----
    { "deftpx", cm_defun, NO_BRACE_ARGS },
  /* The end of the `def' commands. */
  
+ #ifdef JOKES
+   { "email", cm_email, BRACE_ARGS },
+   { "explain", cm_explain, BRACE_ARGS },
+   { "fyi", cm_fyi, BRACE_ARGS },
+   { "host", cm_host, BRACE_ARGS },
+   { "news", cm_news, BRACE_ARGS },
+   { "ifhtml", command_name_condition, NO_BRACE_ARGS },
+ #endif
+ 
    { "display", cm_display, NO_BRACE_ARGS },
    { "dots", cm_dots, BRACE_ARGS },
    { "dmn", do_nothing, BRACE_ARGS },
***************
*** 889,896 ****
--- 903,917 ----
  /* Display the version info of this invocation of Makeinfo. */
  print_version_info ()
  {
+ 
+ #ifdef JOKES
+   fprintf (stderr, "This is GNU Makeinfo version %d.%dj (-joke).\n",
+ 	   major_version, minor_version);
+ #else
    fprintf (stderr, "This is GNU Makeinfo version %d.%d.\n",
  	   major_version, minor_version);
+ #endif
+ 
  }
  
  /* **************************************************************** */
***************
*** 1827,1834 ****
--- 1848,1862 ----
  
    if (!no_headers)
      {
+ 
+ #ifdef JOKES
+       add_word_args ("This is Info file %s, produced by Makeinfo-%d.%dj (-joke) from ",
+ 		     output_filename, major_version, minor_version);
+ #else
        add_word_args ("This is Info file %s, produced by Makeinfo-%d.%d from ",
  		     output_filename, major_version, minor_version);
+ #endif
+ 
        add_word_args ("the input file %s.\n", input_filename);
      }
  
***************
*** 3419,3424 ****
--- 3447,3525 ----
        in_fixed_width_font--;
      }
  }
+ 
+ #ifdef JOKES
+ cm_email (arg)
+      int arg;
+ {
+   extern int printing_index;
+ 
+   if (printing_index)
+     return;
+ 
+   if (arg == START)
+     {
+       in_fixed_width_font++;
+       add_char ('<');
+     }
+   else
+     {
+       add_word (">");
+       in_fixed_width_font--;
+     }
+ }
+ 
+ cm_explain (arg)
+      int arg;
+ {
+   extern int printing_index;
+ 
+   if (printing_index)
+     return;
+ 
+   if (arg == START)
+     {
+       in_fixed_width_font++;
+     }
+   else
+     {
+       add_word (":");
+       in_fixed_width_font--;
+     }
+ }
+ 
+ cm_fyi (arg)
+      int arg;
+ {
+   extern int printing_index;
+ 
+   if (printing_index)
+     return;
+ 
+   if (arg == START)
+     {
+       in_fixed_width_font++;
+       add_char ('"');
+     }
+   else
+     {
+       add_word ("\"");
+       in_fixed_width_font--;
+     }
+ }
+ 
+ cm_news (arg)
+      int arg;
+ {
+   cm_code (arg);
+ }
+ 
+ cm_host (arg)
+      int arg;
+ {
+   cm_code (arg);
+ }
+ #endif
  
  cm_samp (arg)
       int arg;
